tests/unit/src/test/scala/tests/DocumentHighlightLspSuite.scala

```diff
@@ -99,6 +99,47 @@ class DocumentHighlightLspSuite extends BaseLspSuite("documentHighlight") {
       |}""".stripMargin
   )
 
+  check(
+    "local-var",
+    """
+      |object Test {
+      |  def met() = {
+      |    class T1(var <<abc>>: Int) {
+      |      <<ab@@c>> = 4
+      |      def m2: Int = <<abc>> + 2
+      |    }
+      |  }
+      |}""".stripMargin
+  )
+  check(
+    "local-assign",
+    """
+      |object Test {
+      |  def met() = {
+      |    class T1(var <<abc>>: Int) {
```

Comment:
Could you add a second class T2 so that we can see if we don't highlight too much.